### ISSUE TITLE
Arrow keys don't jump to ends of mouse-created selections extending to either end

### ISSUE THREAD

**Issue by pushfoo** [pyglet/pyglet#1074]
### Summary

TL;DR: If you click-drag to either end of a text input's contents in either direction, the opposite direction's arrow key won't jump & end the selection as expected.

It's unclear whether this is platform specific, but it's in the left / right arrow key selection response behavior as described in #932.

Selections created with shift + arrow keys work as expected.

### Versions affected

Tip of `master` + anything after #1072 

### Platform info

Linux, Debian 11.
Python 3.9.2
Gnome on Wayland with XWayland

#### Left Arrow Key

1. Launch `examples/gui/widgets.py`
2. Left-click anywhere before the end of the text
3. Drag the cursor rightward to the very end, making sure to include the last character
4. Release the left mouse button
5. Press the left arrow key
6. Observe:
   * The cursor has not moved
   * The selection has not been deselected

#### Right Arrow Key

1. Launch `examples/gui/widgets.py`
2. Left-click anywhere after the start of the text
3. Drag to cursor the very start, making sure to include the first character
4. Release the left mouse button
5. Press the right arrow key
6. Observe:
   * The cursor has not moved
   * The selection has not been deselected

**Comment by sabadam32** [comment]
@pushfoo tried this on Windows 10 and did not see the issue

**Comment by caffeinepills** [comment]
Closing as this seems resolved when testing in the master branch after testing. If problems still persist, let us know a new way to produce it.
